framework/src/service/ServiceRegistrationCoreInfo.h

```diff
@@ -0,0 +1,100 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef CPPMICROSERVICES_SERVICEREGISTRATIONCOREINFO_H
+#define CPPMICROSERVICES_SERVICEREGISTRATIONCOREINFO_H
+
+#include "cppmicroservices/ServiceInterface.h"
+#include "cppmicroservices/ServiceReference.h"
+#include "cppmicroservices/detail/Threads.h"
+
+#include "BundlePrivate.h"
+#include "Properties.h"
+
+#include <atomic>
+
+namespace cppmicroservices
+{
+
+    /**
+     * \ingroup MicroServices
+     */
+    class ServiceRegistrationCoreInfo : public detail::MultiThreaded<>
+    {
+
+      public:
+        ServiceRegistrationCoreInfo(BundlePrivate* bundle, InterfaceMapConstPtr service, Properties&& props);
+        ~ServiceRegistrationCoreInfo() = default;
+
+        ServiceRegistrationCoreInfo(ServiceRegistrationCoreInfo const&) = delete;
+        ServiceRegistrationCoreInfo& operator=(ServiceRegistrationCoreInfo const&) = delete;
```

Comment:
a `= default;` destructor stops the compiler from generating implicit move operators. Per https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all, the move constructor and assignment need to be defined as `= delete` as well.

see https://en.cppreference.com/w/cpp/language/rule_of_three for more detailed information.

Reply:
This has been changed and fixed